Data-Structures-Trie.md

```diff
@@ -0,0 +1,99 @@
+# Trie
+
+## Introduction to Trie
+
+The word trie is an inflix of the word "re**trie**val", because the trie can find a single word in a dictionary with only a prefix of the word.
+Trie is an efficient data retrieval data structure, using trie, search complexities can be brought to an optimal limit, i.e. length of the string.
+It is a multi-way tree structure useful for storing strings over an alphabet.
+It has been used to store large dictionaries (do not confuse this with the dictionaries of python) of any language, say English, words in spell-checking programs.
+However, the penalty on tries is the storage requirement.
+
+## What is a trie?
+
+A trie is a tree like data structure which stores strings, and helps you find the data associated with that string using the prefix of the string. 
+For example, say you plan on building a dictionary to store strings along with their meanings. You must be wondering why can't I simply use a hash table, to get the information. Yes, you obviously can get information using a hash table, but the worst time complexity for fetching data from a hash table is `O(n)`, where `n` is the number of strings stored. 
```

Comment:
Well, technically it is `O(1)`, however, with tries, further searches work better when your string is growing by size (in an auto-suggest program, for example).